client/components/product-details.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+
+export default class ProductDetails extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      product: {}
+    };
```

Comment:
This should start out with a `null` product. In your `render()` method, you can `return null` if the product isn't loaded yet.

That should prevent the flash of a skeleton while the product data is being loaded.